### ISSUE TITLE
! in url password issues

### ISSUE THREAD

**Issue by admwx7** [systemjs/systemjs#839]
So we've got an admin page setup that requires a username/password to simulate a user during dev, for some reason the company I'm contracted to decided they wanted the password to start with a '!'.  I switched the app over to JSPM and I'm getting the following errors: (password obfuscated with ****)

Error: Script terminated by timeout at:
e@http://admin:!****@localhost/gm-ep/common/jspm_packages/system.js:4:22173
s/<@http://admin:!****@localhost/gm-ep/common/jspm_packages/system.js:4:24595
e@http://admin:!****@localhost/gm-ep/common/jspm_packages/system.js:5:13609
e/</</</<@http://admin:!****@localhost/gm-ep/common/jspm_packages/system.js:5:13681
e/</</</<@http://admin:!****@localhost/gm-ep/common/jspm_packages/system.js:5:15601
 system.js:4:22172
Error: e is undefined
	Error loading http://admin:!****@localhost/gm-ep/common/****@localhost/gm-ep/common/****@localhost/gm-ep/common/****@localhost/gm-ep/common/****@localhost/gm-ep/common/****@localhost/gm-ep/common/****@localhost/gm-ep/common/****@localhost/gm-ep/common/****@localhost/gm-ep/common/****@localhost/gm-ep/common/****@localhost/gm-ep/common/****@localhost/gm-ep/common/****@localhost/gm-ep/common/****@localhost/gm-ep/common/****@localhost/gm-ep/common/****@localhost/gm-ep/common/****@localhost/gm-ep/common/****@localhost/gm-ep/common/****@localhost/gm-ep/common/****@localhost/gm-ep/common/****@localhost/gm-ep/common/****@localhost/gm-ep/common/****@localhost/gm-ep/common/****@localhost/gm-ep/common/****@localhost/gm-ep/common/****@localhost/gm-ep/common/****@localhost/gm-ep/common/****…
Stack trace:
E@http://admin:!****@localhost/gm-ep/common/jspm_packages/system.js:4:2780
e/</</</<@http://admin:!****@localhost/gm-ep/common/jspm_packages/system.js:5:15643

The script eventually times out and fails.  I've just removed the '!' from the password and everything works fine.  Also when running manually replacing the '!' with '%21' in the url works fine, however protractor with browser.get() converts '%21' back to '!' which is why the url has the username/password in it to begin with, for the e2e tests.

**Comment by guybedford** [comment]
If protractor is converting the `%21` is this perhaps a protractor issue?

**Comment by admwx7** [comment]
Going to the url: http://admin:!apassword@localhost/../index.html will
cause the same timeout issue, replacing the ! with %21 will fix the issue
though.  The protractor issue is another bug I'm going to have to post for
them.

**Comment by guybedford** [comment]
@admwx7 can you not just include the %21 in your paths or map config directly here?

**Comment by guybedford** [comment]
I'm must trying to understand exactly where in the pipeline SystemJS would be able to handle this.

**Comment by admwx7** [comment]
I've got a work around in place that works for us, just wanted to make you aware that there is a defect in how the systemjs script is parsing that is causing it to fail.  I looked into protractor, the issue is actually with the url package on npm converting url encoding then parsing, which actually wouldn't be an issue if systemjs wasn't timing out when it saw a valid character in the url.  My first thought was to use encodeURIComponent, but ! is valid and passes through.
